src/twisted/internet/endpoints.py

```diff
@@ -888,6 +1016,22 @@ def cancelRemainingPending(result):
         return d
 
 
+    def _nameResolution(self, host, port):
+        """
+        Resolve the hostname string into a tuple containing the host
+        address.
+
+        @param host: A unicode hostname to resolve.
+
+        @param port: The port to include in the resolution.
+
+        @return: A L{Deferred} that fires with L{_getaddrinfo}'s
+            return value.
+        """
+        return self._deferToThread(self._getaddrinfo, host, port, 0,
+                socket.SOCK_STREAM)
```

Comment:
I'm not sure that it makes sense to allow overriding `deferToThread` and `getaddrinfo` individually, instead of just overriding this function.